pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -374,6 +370,13 @@ func (r *ReconcileCredentialsRequest) Reconcile(ctx context.Context, request rec
 	}()
 
 	mode, conflict, err := utils.GetOperatorConfiguration(r.Client, logger)
+
+	stsDetected := false
```

Comment:
We do this block of code as well inside of the AWS actuator - perhaps just pull it out and take the detected bool itself in the actuator constructor?

Reply:
Not sure if code below looked like this:
```go
stsFeatureGateEnabled := r.Actuator.STSFeatureGateEnabled()
```
when the comment was made, but this seems to answer that, at least enough to not mess with the actuator initialization more...